gdk/common/recipe_file/CaseInsensitiveRecipe.py

```diff
@@ -0,0 +1,54 @@
+from requests.structures import CaseInsensitiveDict
+from gdk.common.recipe_file.Recipe import Recipe
+
+
+class CaseInsensitiveRecipe:
+    def __convert_nested_dict(self, recipe: CaseInsensitiveDict) -> CaseInsensitiveDict:
+        for key, value in recipe.items():
+            if isinstance(value, dict):
+                recipe.update({key: self.__convert_nested_dict(CaseInsensitiveDict(value))})
+            elif isinstance(value, list):
+                recipe.update(
+                    {key: [self.__convert_nested_dict(CaseInsensitiveDict(val)) for val in value if isinstance(val, dict)]}
+                )
+        return recipe
+
+    def __convert_nested_CaseInsensitiveDict(self, recipe: dict) -> dict:
+        for key, value in recipe.items():
+            if isinstance(value, CaseInsensitiveDict):
+                recipe.update({key: self.__convert_nested_CaseInsensitiveDict(dict(value))})
+            elif isinstance(value, list):
+                recipe.update(
+                    {
+                        key: [
+                            self.__convert_nested_CaseInsensitiveDict(dict(val))
+                            for val in value
+                            if isinstance(val, CaseInsensitiveDict)
+                        ]
+                    }
+                )
+        return recipe
+
+    def __get(self, original: dict) -> CaseInsensitiveDict:
+        __recipe = CaseInsensitiveDict(original)
+        self.__convert_nested_dict(__recipe)
+        return __recipe
+
+    def __get_as_dict(self, cir: CaseInsensitiveDict) -> dict:
+        __recipe = dict(cir)
```

Comment:
why the `__` I think there is no need in this case. There is no need to do name mangling here and in most of the other methods. This one should be `recipe` and the names should use single underscore just to indicate they are private. In python using __ doesn't mean you won't be able to access it but that the interpreter would tweak the name when using double `__`